core/src/main/scala/cats/collections/Heap.scala

```diff
@@ -130,14 +130,20 @@ sealed abstract class Heap[A] {
     Heap.heapify(a)
 
   /**
-   * Remove the min element from the heap (the root).
+   * Remove the min element from the heap (the root) and return it along with the updated heap.
    * Order O(log n)
    */
-  def remove(implicit order: Order[A]): Heap[A] = this match {
-    case Branch(_, l, r) => bubbleRootDown(mergeChildren(l, r))
-    case Leaf()          => Leaf()
+  def pop(implicit order: Order[A]): (Heap[A], Option[A]) = this match {
+    case Branch(m, l, r) => (bubbleRootDown(mergeChildren(l, r)), Some(m))
+    case Leaf()          => (this, None)
   }
 
+  /**
+   * Remove the min element from the heap (the root).
+   * Order O(log n)
+   */
+  def remove(implicit order: Order[A]): Heap[A] = this.pop._1
```

Comment:
Can we keep the old implementation to save some allocations on this variant?  I’m less interested in code reuse than performance with libraries we hope to be adopted widely.

Reply:
ok

Reply:
I originally implemented `heap` with performance and immutability in mind, be careful with changes that change those properties.